src/mesh/rbf.f90

```diff
@@ -0,0 +1,597 @@
+TYPE(tSide),POINTER    :: Side
+INTEGER                :: i,j,nAns
+REAL,ALLOCATABLE       :: RefCoordinates(:,:)            ! Coordinates of boundary points
+REAL,ALLOCATABLE       :: RBFMatrix(:,:)                 ! Matrix of the linear system
+REAL,ALLOCATABLE       :: RBFRHS(:,:)
+INTEGER                :: nBP                            ! Number of boundary points considered
+REAL,ALLOCATABLE       :: Vdm_SurfBiLinear(:,:),Vdm_VolTriLinear(:,:)
+REAL,ALLOCATABLE       :: D_tmp(:,:,:)
+INTEGER,ALLOCATABLE    :: QuadMapInvLinear(:,:)
+INTEGER,ALLOCATABLE    :: HexaMapInvLinear(:,:,:)
+INTEGER,ALLOCATABLE    :: QuadMapLinear(:,:)
+INTEGER,ALLOCATABLE    :: HexaMapLinear(:,:)
+REAL                   :: xTriLinear(3,BoundaryOrder**3),xBiLinear(3,BoundaryOrder**2)
+REAL                   :: xCornerVol(1:3,8),xCornerSurf(1:3,4)
+REAL                   :: dist,x(3),xTmp(3),rbfvalue
+INTEGER                :: iBP
+REAL                   :: xMin,xMax,yMin,yMax,xBary(3)
+!===================================================================================================================================
+
+WRITE(UNIT_StdOut,'(132("-"))')
+WRITE(UNIT_stdOut,'(A)') ' RBF VOLUME CURVING'
+
+! Box where curving should be done
+xMin = xlim(1,iRBFBox)
+xMax = xlim(2,iRBFBox)
+yMin = ylim(1,iRBFBox)
+yMax = ylim(2,iRBFBox)
+
+! Build 2D Vandermonde from corner nodes to bi-linear side nodes
+CALL getQuadBasis(1,BoundaryOrder,Vdm_SurfBiLinear,D_tmp)
+DEALLOCATE(D_tmp)
+
+! Build 3D Vandermonde from corner nodes to tri-linear side nodes
+CALL getHexaBasis(1,BoundaryOrder,Vdm_VolTriLinear,D_tmp)
+DEALLOCATE(D_tmp)
+
+! Build mapping between 1D and tensor product ordering for a bi-linear surface
+CALL getBasisMappingQuad(1,nAns,QuadMapLinear,QuadMapInvLinear)
+DEALLOCATE(QuadMapLinear)
+
+! Build mapping between 1D and tensor product ordering for a bi-linear surface
+CALL getBasisMappingHexa(1,nAns,HexaMapLinear,HexaMapInvLinear)
+DEALLOCATE(HexaMapLinear)
+
+
+!--------------------------------------------------------------------------------------------------------!
+! Count number of control points, needed to allocate space etc.
+!--------------------------------------------------------------------------------------------------------!
+
+! Reset the tmp variable later used to mark already visited control points
+Elem=>FirstElem
+DO WHILE(ASSOCIATED(Elem))
+  Side=>Elem%firstSide
+  DO WHILE(ASSOCIATED(Side))
+    IF(.NOT.ASSOCIATED(Side%BC)) THEN
+      ! Go through all 4 corner nodes
+      Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 0
+    ELSE IF(Side%BC%BCType.EQ.1) THEN
+      ! Go through all 4 corner nodes
+      Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 0
+      Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 0
```

Comment:
I do not like this is writing on `tmp` again, especially given #8. We come up with a better way but maybe a simple rename will suffice.

Reply:
After the discussion, we keep the tmp field, which is there exactly for something like this: short-term information with a local scope.